rule/rule.go

```diff
@@ -143,44 +165,78 @@ func (r *Rule) UnmarshalJSON(raw []byte) error {
 		ID             string         `json:"id"`
 		Version        string         `json:"version"`
 		Description    string         `json:"description"`
-		Match          *Match         `json:"match"`
 		Authenticators []Handler      `json:"authenticators"`
 		Authorizer     Handler        `json:"authorizer"`
 		Mutators       []Handler      `json:"mutators"`
 		Errors         []ErrorHandler `json:"errors"`
 		Upstream       Upstream       `json:"upstream"`
-		matchingEngine MatchingEngine
+
+		RawMatch json.RawMessage `json:"match"`
+		Match    URLProvider
 	}
 
 	transformed, err := migrateRuleJSON(raw)
 	if err != nil {
-		return err
+		return errors.WithStack(err)
 	}
 
-	if err := errors.WithStack(json.Unmarshal(transformed, &rr)); err != nil {
-		return err
+	if err := json.Unmarshal(transformed, &rr); err != nil {
```

Comment:
I couldn't find a dedicated test for this new functionality, did I miss it?

Reply:
You're right, that was not covered. I added dedicated Unmarshal tests.